x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_panel.tsx

```diff
@@ -497,7 +534,11 @@ export function SuggestionPanel({
           role="list"
           tabIndex={0}
         >
-          {changesApplied ? renderSuggestionsUI() : renderApplyChangesPrompt()}
+          {changesApplied
+            ? activeData
+              ? renderSuggestionsUI()
+              : renderSuggestionsLoadingState()
+            : renderApplyChangesPrompt()}
```

Comment:
Wait for active data before starting suggestion render. Otherwise, initial suggestion render is interrupted by a re-render when active data arrives, making it difficult to actually track when the suggestions are finished.

There are other ways we could approach this if we don't like this one.